crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -336,24 +342,33 @@ async fn put_handler(
     req: Json<PutReq>,
     mut db_conn: DbConnection,
     user: User,
+    #[cfg(feature = "high-performance-mode")] tenant: Tenant,
     tenant_config: TenantConfig,
 ) -> superposition::Result<HttpResponse> {
     let tags = parse_config_tags(custom_headers.config_tags)?;
-    db_conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+    let mut version_id = 0;
+    let resp = db_conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
         let put_response = put(req, transaction_conn, true, &user, &tenant_config)
             .map_err(|err: superposition::AppError| {
                 log::info!("context put failed with error: {:?}", err);
                 err
             })?;
-        let version_id = add_config_version(&state, tags, transaction_conn)?;
+        version_id = add_config_version(&state, tags, transaction_conn)?;
         let mut http_resp = HttpResponse::Ok();
 
         http_resp.insert_header((
             AppHeader::XConfigVersion.to_string(),
             version_id.to_string(),
         ));
         Ok(http_resp.json(put_response))
-    })
+    });
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "high-performance-mode")] {
+            let DbConnection(mut conn) = db_conn;
+            put_config_in_redis(version_id, state, tenant, &mut conn).await?;
+        }
+    }
+    resp
 }
 
 fn override_helper(
```

Comment:
Favor:
A very similar function `put` is also exists, can you remove this and make the necessary changes?

Reply:
Yeah I noticed this, sure let me change this.